openstack-api-charm-creation-guide.md

```diff
@@ -0,0 +1,240 @@
+# Guide to creating a basic Openstack API service charm
+
+This guide will walk the creation of a basic charm for the Openstack
+Congress service. [Congress Info](https://wiki.openstack.org/wiki/Congress).
+The charm will use prewritten Openstack [layers and interfaces](https://github.com/openstack-charmers)
+Once the charm is written it will be compose using [charm tools](https://github.com/juju/charm-tools/).
+
+The Congress service needs to register endpoints with Keystone. It needs a service username and password and it also needs a MySQL backend to store its schema.
+
+# Create the skeleton charm
+
+## Create charm directories
+
+Firstly create a directory for the new charm and manage the charm with git.
+
+```bash
+mkdir -p congress/charm
+cd congress
+git init
+```
+The top layer of this charm is the Congress specific code this code will live in the charm subdirectory.
+
+```mkdir -p charm/{reactive,lib}```
+
+## Describe the Service and required layer(s)
+
+The new charm needs a basic metadata.yaml to describe what service the charm provides. Edit charm/metadata.yaml
```

Comment:
Some of the files that need to be created (below) need to specify that they're under the charm/ directory.